EmailService/EmailService.Business/Commands/ModuleSetting/CheckSmtpCommand.cs

```diff
@@ -0,0 +1,114 @@
+using LT.DigitalOffice.Models.Broker.Enums;
+using LT.DigitalOffice.Models.Broker.Responses.TextTemplate;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Options;
+
+
+namespace LT.DigitalOffice.EmailService.Business.Commands.ModuleSetting
+{
+  public class CheckSmtpCommand : ICheckSmtpCommand
+  {
+    private readonly IAccessValidator _accessValidator;
+    private readonly IResponseCreator _responseCreator;
+    private readonly ICheckSmtpRequestValidator _validator;
+    private readonly ISmtpSettingsRepository _repository;
+    private readonly IMemoryCache _cache;
+    private readonly IOptions<MemoryCacheConfig> _cacheOptions;
+    private readonly EmailSender _emailSender;
+    private readonly ITextTemplateService _textTemplateService;
+    private readonly ITextTemplateParser _parser;
+
+    private int GenerateCode()
+    {
+      Random rnd = new Random();
+      return rnd.Next(10000, 100000);
+    }
+
+    private async Task<bool> SendCodeAsync(string email, string locale, List<string> errors, string code, SmtpInfo smtpInfo)
+    {
+      IGetTextTemplateResponse textTemplate = await _textTemplateService.GetAsync(TemplateType.SmtpCheck, locale, errors);
+
+      if (textTemplate is null)
+      {
+        return false;
+      }
+
+      string parsedText = _parser.Parse(
+        new Dictionary<string, string> { { "Code", code } },
+        textTemplate.Text);
+
+      return await _emailSender.SendCodeAsync(receiver: email, subject: textTemplate.Subject, text: parsedText, smtpInfo: smtpInfo);
+    }
+
+    public CheckSmtpCommand(
+      IAccessValidator accessValidator,
+      IResponseCreator responseCreator,
+      ICheckSmtpRequestValidator validator,
+      ISmtpSettingsRepository repository,
+      IMemoryCache cache,
+      IOptions<MemoryCacheConfig> cacheOptions,
+      EmailSender emailSender,
+      ITextTemplateService textTemplateService,
+      ITextTemplateParser parser)
+    {
+      _accessValidator = accessValidator;
+      _responseCreator = responseCreator;
+      _validator = validator;
+      _repository = repository;
+      _cache = cache;
+      _cacheOptions = cacheOptions;
+      _emailSender = emailSender;
+      _textTemplateService = textTemplateService;
+      _parser = parser;
+    }
+
+    public async Task<OperationResultResponse<bool>> ExecuteAsync(CheckSmtpRequest request)
+    {
+      bool isFirstCheck = !await _repository.DoesExistAnyAsync();
+
+      if (!isFirstCheck && !await _accessValidator.IsAdminAsync())
+      {
+        return _responseCreator.CreateFailureResponse<bool>(HttpStatusCode.Forbidden);
+      }
+
+      if (!_validator.ValidateCustom(request, out List<string> errors))
+      {
+        return _responseCreator.CreateFailureResponse<bool>(HttpStatusCode.BadRequest, errors);
+      }
+
+      string casheValue = isFirstCheck ? request.GetHashCode().ToString() : request.SmtpInfo.GetHashCode().ToString();
+      string code = GenerateCode().ToString();
```

Comment:
Why do tou use this method. You can write 
string code = new Random().Next(10000, 100000).ToString();

Reply:
SOLID. Single responsibility and open/closed. "Command" should not generate code, this is not it's job. And what if someday we want to change simple code to difficult password with letters and other symbols?

Reply:
no more code-generating